packages/hardhat-core/src/internal/hardhat-network/provider/utils/makeFakeSignature.ts

```diff
@@ -0,0 +1,30 @@
+import util from "util";
+
+import {
+  AccessListEIP2930Transaction,
+  FeeMarketEIP1559Transaction,
+  TxData,
+} from "@nomicfoundation/ethereumjs-tx";
+import { Address } from "@nomicfoundation/ethereumjs-util";
+
+import { createNonCryptographicHashBasedIdentifier } from "../../../util/hash";
+
+// Produces a signature with r and s values taken from a hash of the inputs.
+export function makeFakeSignature(
+  tx: TxData | AccessListEIP2930Transaction | FeeMarketEIP1559Transaction,
+  sender: Address
+): {
+  v: number;
+  r: number;
+  s: number;
+} {
+  const hashDigest = createNonCryptographicHashBasedIdentifier(
+    Buffer.from(`${util.inspect(sender)}${util.inspect(tx)}`)
```

Comment:
Instead of `utill.inspect` we should use a proper list of values. This list is slightly different for each transaction type, so I'm not sure if we should have a separate method for each, or just receive an array of `any`s.

Reply:
Addressed in https://github.com/NomicFoundation/hardhat/pull/3175/commits/2c7dfced2963b37662a5d1d6443ddf1234b798f3